pkg/nmi/server/server.go

```diff
@@ -70,13 +88,13 @@ func NewServer(micNamespace string, blockInstanceMetadata bool) *Server {
 		MICNamespace:          micNamespace,
 		BlockInstanceMetadata: blockInstanceMetadata,
 		Reporter:              reporter,
+		PodClient:             podClient,
+		PodObjChannel:         podObjCh,
 	}
 }
 
 // Run runs the specified Server.
 func (s *Server) Run() error {
```

Comment:
>Run [](start = 17, length = 3)

Nit: Can't this method be called by the nmi/main.go, which in turn runs the linux or windows method to handle routing. We can name those as routing_windows.go and routing_linux.go, and the method could be named as ConfigureRouting. Having two server is kind of confusing now. I feel that way we have a clear line of calls, i.e. the client (main.go) call the server run(), and the server run() configures routing and then listens to the endpoints.

Reply:
Andrey also gives very similar comments like this. I will check if I can implement this as you suggested.

Reply:
I create a interface RedirectorInt in this file and use two new files redirector_windows.go and rifirector_linux.go to impleemnt the function in this interface. In this way, we do not need the two servers. And client will call the server.run, which is the same as before.